docs/docs/dev_guide/plugin_dev_guide.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: default
+title: Plugin Developer Guide Documentation
+parent: Developer's Guide to AaC
+nav_order: 5
+---
+
+# Plugin Developer Guide Documentation
+
+## Configuration of the Environment
+
+### AaC Setup (Prerequisites)
+
+When performing the steps listed in the [Developer Guide](./index.md). Please make sure that AaC Plugin is executable with the execution of the below command:
+
+`aac version`
+
+If this command is executable please continue with this guide for developing plugins for AaC. 
+
+## AaC Generating Plugins (Gen-Plugin Command)
+
+AaC comes pre-packaged with a plugin called `Gen-Plugin` which will take an AaC architecture file describing a plugin and then generate the necessary code stubs and python project infrastructure for the user to begin implementing their new plugin.
+
+### How to Use Gen-Plugin and Output
+
+The `gen-plugin` command can be executed with the following syntax with the `aac` plugin:
+`aac gen-plugin ./path/to/architecture_file.yaml`
+#### Input of Gen-Plugin
+
+Plugins can be created with the first-party plugin `gen-plugin`. This can be executed from a shell/terminal, with the following example command:
+`aac gen-plugin -help`
+The above command will output the help documentation for the `gen-plugin` command with example input.
+
+The `gen-plugin` command takes input from an architecture file and will produce an output of templated plugin files that can be used to install the first/third party plugin that is being developed. An example plugin file is below:
+> *The below example can be found under this path -> `/python/model/plugin/plugin.yaml`*
+
+```yaml
+model:
+  name: test-plugin
+  description: first-party is a test plugin
+  behavior:
+    - name: test-plugin-command
+      type: command
+      description: Test plugin generation
+      input:
+        - name: architecture_file
+          type: file
+          python_type: str
+          description: An architecture-as-code file.
+      acceptance:
+        - scenario: Test some stuff
+          given:
+            - The {{test-plugin-command.input.architecture_file}} represent a valid system architecture.
+          when:
+            - The command is run
+          then:
+            - Then stuff happens
+---
+schema:
+  name: TestPluginData
+  fields:
+    - name: value1
+      type: string
+    - name: value2
+      type: string
+  validation:
+    - name: Required fields are present
+      arguments:
+        - value1
+```
+
+#### Output of Gen-Plugin
+
+The output of the Gen-Plugin command will be some corresponding plugin files that are templated from the `yaml` file that was passed in to `gen-plugin`
+
+The generated files are placed into the original directory containing the AaC architecture file.
+
+- `__init.py__`
+- `plugin_name_impl.py`
+- `plugin_architecture_file.yaml`
+- `jinja` User-made `jinja` template files can be stored here for plugins that generate artifacts.
+
+The folder structure will look something like this:
```

Comment:
We should point out that this is only the folder structure for plugins that are generated as third-party plugins.